tests/unit/utils/test_pydantic_utils.py

```diff
@@ -0,0 +1,96 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from typing import Dict, Optional
+
+from pydantic import BaseModel
+
+from zenml.utils import pydantic_utils
+
+
+def test_update_model_works():
+    """Tests that updating a Pydantic model using our util function works."""
+
+    class TestModel(BaseModel):
+        a: int
+        b: int
+
+    original = TestModel(a=1, b=2)
+    update = TestModel(a=3, b=4)
+    updated = pydantic_utils.update_model(original, update)
+    assert updated.a == 3
```

Comment:
I guess this could include something that tests the `exclude_unset` functionality. For example, make b an `Optional[int]` and don't set it on the update model. In that case, the original value should be preserved instead of setting the value to `None`. Or give `b` a default value that doesn't get set, tests the same thing.